docs/how-to-guides/how-to-integrate-autoware-with-diff-drive-vehicle.md

```diff
@@ -0,0 +1,59 @@
+# How to integrate Autoware with differential drive vehicle
+
+## 1. Integrate Autoware with differential drive vehicle
+
+Currently, Autoware assumes the vehicle to follow an Ackermann kinematic model that uses Ackermann steering.
+Thus, the Autoware interfaces adopt the Ackermann commands format for the output of the control module (see [the definition of AckermannControlCommands.idl](https://gitlab.com/autowarefoundation/autoware.auto/autoware_auto_msgs/-/blob/master/autoware_auto_control_msgs/msg/AckermannControlCommand.idl) for details).
+
+However, it is possible to integrate Autoware with a vehicle that follows a differential drive kinematic model, which is especially common for small mobile robots.
+
+## 2. Procedure
+
+This section briefly explains one of the ways to use Autoware for differential drive vehicles.
+One simple way is to create a `vehicle_interface` package that translates the Ackermann command to the differential drive command.
+Here are two points that you need to consider:
+
+- Create `vehicle_interface` package for differential drive vehicle
+- Set an appropriate `wheel_base`
+
+### 2.1 Create `vehicle_interface` package for differential drive vehicle
+
+Ackermann command in Autoware mainly consists of two main control inputs:
+
+- steering angle ($\omega$)
+- velocity ($v$)
+
+while the typical differential drive command consists of the following inputs:
+
+- left wheel velocity ($v_l$)
+- right wheel velocity ($v_r$)
+
+For example, the Ackermann command can be converted to the differential drive command with the following equations:
+
+$$
+v_l = v - \frac{l\omega}{2},
+v_r = v + \frac{l\omega}{2}
+$$
+
+where $l$ denotes wheel tread.
+
+For general requirements for `vehicle_interface` package, please refer to [the description of `vehicle_interface`](https://autowarefoundation.github.io/autoware-documentation/main/design/autoware-interfaces/components/vehicle-interface/) for detail.
+
+### 2.2 Set an appropriate `wheel_base`
+
+Differential drive robot does not necessarily have front and rear wheels, which means that the wheel base may not exist, although Autoware currently requires `wheel_base` to be given in `vehicle_info.param.yaml`.
+Thus, you need to set a pseudo value for `wheel_base`.
+
+Recommended value for `wheel_base` depends on the size of your vehicle. Setting it as the same value as `wheel_tread` would be one of the choices.
+
+Note that setting the `wheel_base` too large may end up in unflexible control. Also, current autoware.universe is not guaranteed to work when the value is set too small.
```

Comment:
> Is hard to control okay to use instead of inflexible control?

Thanks for the suggestion. `hard to control` can also reflect what I meant here, but still a bit vague. What do you think about `the vehicle's motion may be restricted` instead?

> I feel that not guaranteed to work is too vague. Roughly what kind of problems may occur eg: Autoware may throw a runtime error?

Unfortunately, Autoware does not necessarily throw a runtime error even when small wheel_base does harmful things. One trouble we've had was that the vehicle couldn't follow the calculated path properly and went out of the road. 

But I agree with you that `guaranteed to work` is a bit vague. Do you have any other suggestions?

Reply:
> Thanks for the suggestion. hard to control can also reflect what I meant here, but still a bit vague. What do you think about the vehicle's motion may be restricted instead?

The problem is that someone will ask what `restricted` means. Would it be accurate to say that the vehicle cannot be controlled with precision?

> But I agree with you that guaranteed to work is a bit vague. Do you have any other suggestions?

I think it is better to give concrete examples where possible so that users/developers have a full understanding of any potential risks/issues. How about this?

`Conversely, if the `wheel_base` value is too small then the vehicle may behave unexpectedly. For example, the vehicle may drive beyond the bounds of a calculated path.`

Reply:
The former part:
`cannot be controlled with precision` is not correct. The problem of setting the `wheel_base` too large is, for example, if you set `wheel_base` to 10m, Autoware has to calculate a trajectory that 10m long vehicle always stays within the drivable area, which will highly restrict the vehicle's motion.

How about this?
`If wheel_base is too large, Autoware will calculate excessively conservative trajectory in the planning module, as it assumes that the vehicle is too long`

The latter part:
Thanks for the suggestion. I reflected the comments in https://github.com/autowarefoundation/autoware-documentation/pull/125/commits/907774cc3e3dd78c59b088bf38e3f33cc1bba761

Reply:
That would make sense! Thank you for the suggestion.
https://github.com/autowarefoundation/autoware-documentation/pull/125/commits/4859401924f3086f6cc285a18ce6742ab4e13278